src/components/ImgEditor/useImgEditor.ts

```diff
@@ -1,98 +1,81 @@
-import { useErrorContext } from "contexts/ErrorContext";
 import { useModalContext } from "contexts/ModalContext";
 import { MouseEventHandler } from "react";
 import { DropzoneOptions } from "react-dropzone";
-import { FieldValues, useController, useFormContext } from "react-hook-form";
+import { useController, useFormContext } from "react-hook-form";
 import { Path } from "react-hook-form";
 import ImgCropper from "./ImgCropper";
 import { ImgLink, Props } from "./types";
 
-type Key = keyof ImgLink;
-const fileKey: Key = "file";
-const previewKey: Key = "preview";
-const precropFileKey: Key = "precropFile";
+type Base = { [index: string]: ImgLink };
 
-type FilePath = `${string}.${typeof fileKey}`;
-
-export default function useImgEditor<T extends FieldValues, K extends Path<T>>({
+export default function useImgEditor<T extends Base, K extends Path<T>>({
   name,
   aspect,
   accept,
 }: Props<T, K>) {
-  const filePath: any = `${String(name)}.${fileKey}`;
-  const previewPath: any = `${String(name)}.${previewKey}`;
-  const precropFilePath: any = `${String(name)}.${precropFileKey}`;
+  const path = (sub: Path<ImgLink>): Path<T> => `${name}.${sub}` as any;
 
-  const { setValue, watch } = useFormContext<T>();
+  const { setValue, watch, resetField, trigger } = useFormContext<T>();
   const { showModal } = useModalContext();
-  const { handleError } = useErrorContext();
   const {
     field: {
       value: currFile = new File([], "default file"),
       onChange: onFileChange,
       ref,
     },
-  } = useController<Record<string, ImgLink>, FilePath>({
-    name: filePath,
+  } = useController<Base, `${string}.file`>({
+    name: `${name}.file`,
   });
 
-  const { publicUrl, preview, precropFile }: ImgLink = watch(name as any);
+  const { publicUrl, preview }: ImgLink = watch(name as any);
   const isInitial = preview === publicUrl;
   const noneUploaded = !publicUrl && !preview;
 
   const onDrop: DropzoneOptions["onDrop"] = (files: File[]) => {
     const newFile = files[0];
-    if (newFile) {
-      //preview & crop valid formats only
-      if (!!accept.find((x) => x === newFile.type)) {
-        const preview = URL.createObjectURL(newFile);
-        showModal(ImgCropper, {
-          preview,
-          aspect,
-          type: newFile.type,
-          onSave(blob) {
-            handleCropResult(blob, newFile);
-          },
-        });
-      }
-      onFileChange(newFile);
-      setValue(precropFilePath, newFile as any, { shouldValidate: true });
+    if (!newFile) return;
+
+    // set the file, and validate immediately
```

Comment:
form validate mode is `onSubmit`, so initial uploads doesn't give error - so trigger validation manually rather than making the whole form validation `onChange`